src/OpenTelemetry/BaseExportProcessor.cs

```diff
@@ -44,7 +45,17 @@ public sealed override void OnStart(T data)
         }
 
         /// <inheritdoc />
-        public abstract override void OnEnd(T data);
+        public sealed override void OnEnd(T data)
+        {
+            if (data is Activity activity && activity.ActivityTraceFlags == ActivityTraceFlags.None)
```

Comment:
Not a blocker. We might want to revisit this part and see if we could do some optimization post GA.

Reply:
* I think it might be better to check for `Recorded` as opposed to `None` here, just in case any other flags are added to `ActivityTraceFlags` in the future.

* I am strongly opposed to this pattern of having generic classes with type-specific logic hidden inside. If we need type-specific logic we should do something like: `ActivityBaseExportProcessor : BaseExportProcessor<Activity>`.

Reply:
@utpilla Can you open an issue to track fixing this? It should be easy to fix, but we need to make sure an issue is open so that this is not lost.

Reply:
I have created the issue #1595 to track this.